packages/headless/src/controllers/knowledge/generatedAnswer/headless-knowledge-generated-answer.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  answerApi,
+  fetchAnswer,
+  selectAnswer,
+  StateNeededByAnswerAPI,
+} from '../../../api/knowledge/stream-answer-api';
+import {CoreEngine} from '../../../app/engine';
+import {SearchEngine} from '../../../app/search-engine/search-engine';
+import {
+  resetAnswer,
+  updateAnswerConfigurationId,
+} from '../../../features/generated-answer/generated-answer-actions';
+import {
+  ConfigurationSection,
+  DebugSection,
+  GeneratedAnswerSection,
+  SearchSection,
+} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+import {
+  buildCoreGeneratedAnswer,
+  GeneratedAnswer,
+  GeneratedAnswerAnalyticsClient,
+  GeneratedAnswerProps,
+  GeneratedResponseFormat,
+} from '../../core/generated-answer/headless-core-generated-answer';
+
+export interface KnowledgeGeneratedAnswer extends GeneratedAnswer {
+  /**
+   * Resets the last answer.
+   */
+  reset(): void;
+}
+
+interface KnowledgeGeneratedAnswerProps extends GeneratedAnswerProps {}
+
+export interface SearchAPIGeneratedAnswerAnalyticsClient
+  extends GeneratedAnswerAnalyticsClient {}
+
+export type EngineNeededForKnowledgeGeneratedAnswer = CoreEngine<
+  GeneratedAnswerSection &
+    ConfigurationSection &
+    SearchSection &
+    DebugSection & {answer: ReturnType<typeof answerApi.reducer>}
+>;
```

Comment:
I don't think this type is necessary.  Its only used as a cast in the test file

Reply:
quite right, thanks !